### PR TITLE
[CSL-2597] Add to favorites button

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

**Review by esezen — COMMENTED** [review]
This is looking great. Can you add `onAddToFavoritesClick` to `functionStrings` in `utils`?
image

Also do you mind reaching out to Mariano to see if he is happy with the size of the icon? We don't need to wait for his response, we can merge this and change the size in a different PR if needed

File: src/components/CioQuiz/index.tsx [hunk 1]
```diff
@@ -56,6 +58,7 @@ export default function CioQuiz(props: IQuizProps) {
     getResetQuizButtonProps,
     getSelectInputProps,
     customClickItemCallback: !!resultsPageOptions?.onQuizResultClick,
+    customAddToFavoritesCallback: !!resultsPageOptions?.onAddToFavoritesClick,
```
Comment (mocca102) [inline src/components/CioQuiz/index.tsx:61]: Why add `customAddToFavoritesCallback` to the customer facing interface? I'm not opposed to using it just why return it in the hook? I think we can remove it along with `customClickItemCallback`

**Review by mocca102 — CHANGES_REQUESTED** [review]
Can we add another SVG with a filled heart when add_to_favorites fire and only fire the event if the favorites icon isn't active/clicked already. Basically like a toggle since currently the icon doesn't change on click and it looks as if it's not working

File: src/components/CioQuiz/index.tsx (hunk 1 above)
Comment (mocca102) [reply src/components/CioQuiz/index.tsx:61]: Ignore this please

File: src/components/ResultFavoritesButton/ResultFavoritesButton.tsx [hunk 2]
```diff
@@ -8,14 +8,22 @@ interface ResultFavoritesButtonProps {
 }
 
 export default function ResultFavoritesButton(props: ResultFavoritesButtonProps) {
+  const [isFavorited, setIsFavorited] = useState(false);
   const { item, price } = props;
   const { getAddToFavoritesButtonProps } = useContext(QuizContext);
 
+  const clickHandler = () => {
```
Comment (mocca102) [inline src/components/ResultFavoritesButton/ResultFavoritesButton.tsx:15]: Lets not have any logic in presentational components and have the logic in hooks

File: src/components/ResultFavoritesButton/ResultFavoritesButton.tsx [hunk 3]
```diff
@@ -8,14 +8,22 @@ interface ResultFavoritesButtonProps {
 }
 
 export default function ResultFavoritesButton(props: ResultFavoritesButtonProps) {
+  const [isFavorited, setIsFavorited] = useState(false);
```
Comment (mocca102) [inline src/components/ResultFavoritesButton/ResultFavoritesButton.tsx:11]: Lets not have any logic in presentational components and have the logic in hooks

**Comment by mocca102** [comment]
Thanks for making the changes, looking way better. Just a couple of things
- Misspelling in file name useAddToFavoritesbuttonProps -> useAddToFavoritesButtonProps
- Conversion event should only fire if favorites button isn't already clicked
- Not sure if it's this PR but single result card is misaligned

**Review by mocca102 — CHANGES_REQUESTED** [review]
Thanks for making the changes, looking way better. Just a couple of things

- Misspelling in file name useAddToFavoritesbuttonProps -> useAddToFavoritesButtonProps
- Conversion event should only fire if favorites button isn't already clicked
- Not sure if it's this PR but single result card is misaligned
Screenshot 2023-08-03 at 4 22 54 PM

**Comment by mocca102** [comment]
Ignore the visual bug. Enes fixed it in another PR @stanlp1

**Review by mocca102 — APPROVED** [review]
LGTM!

**Pull request merged**